### PR TITLE
fix(SUP-22436): playlist thumb are not loading when loadThumbnailWithKs is set

### PR DESCRIPTION
**Pull request by shalommeoded**
The issue is that thumbnailUrl is set to the first parameter that is declared, just changeing the order resolves this. 

Solves SUP-22436

### CODE REVIEW

File: modules/KalturaSupport/components/playlistAPI.js [hunk 1]
```diff
@@ -397,7 +397,7 @@
 				title = kWidget.sanitize( title );
 				description = kWidget.sanitize( description );
 
-				var thumbnailUrl = item.thumbnailUrl || customData.thumbUrl || this.getThumbUrl(item);
+				var thumbnailUrl = this.getThumbUrl(item) || item.thumbnailUrl || customData.thumbUrl;
```
Comment (OrenMe) [inline modules/KalturaSupport/components/playlistAPI.js:400]: there is a reason for this order, you want it to take it from item or custom level before falling back to default.

Comment (OrenMe) [reply modules/KalturaSupport/components/playlistAPI.js:400]: If issue is KS we need to enforce the `loadThumbnailWithKs` settings option here as well.

Comment (OrenMe) [reply modules/KalturaSupport/components/playlistAPI.js:400]: or maybe handle the item.thumbnailUrl and customData.thumbUrl in `this.getThumbUrl` call

File: modules/KalturaSupport/components/playlistAPI.js [hunk 2]
```diff
@@ -397,15 +397,15 @@
 				title = kWidget.sanitize( title );
 				description = kWidget.sanitize( description );
 
-				var thumbnailUrl = this.getThumbUrl(item) || item.thumbnailUrl || customData.thumbUrl;
+				var thumbnailUrl = item.thumbnailUrl || customData.thumbUrl || this.getThumbUrl(item);
```
Comment (OrenMe) [inline modules/KalturaSupport/components/playlistAPI.js]: same for custom data, so you can make this generic.

Comment (shalommeoded) [reply modules/KalturaSupport/components/playlistAPI.js]: Can you spesify?

Comment (OrenMe) [reply modules/KalturaSupport/components/playlistAPI.js]: `customData.thumbUrl`

File: modules/KalturaSupport/components/chapters/chapters.js [hunk 3]
```diff
@@ -481,7 +481,9 @@
 				var customData = item.partnerData ? JSON.parse(item.partnerData) : {};
 				var title = item.title || customData.title;
 				var description = item.description || customData.desc;
-				var thumbnailUrl = item.thumbnailUrl || customData.thumbUrl || _this.getThumbUrl(item);
+				var thumbnailUrl = mw.getConfig('loadThumbnailWithKs')
```
Comment (OrenMe) [inline modules/KalturaSupport/components/chapters/chapters.js:484]: this is incorrect! you are now basically making the item.thumbnailUrl and customData.thumbUrl obsolete if the flag is set.
You need to take the respective url and add to it a ks if flag is set and not use the `getThumbUrl` global url settings.

Comment (shalommeoded) [reply modules/KalturaSupport/components/chapters/chapters.js]: Done, also no need to add this logic to the chapters plugin, KS is already added.

File: modules/KalturaSupport/components/playlistAPI.js [hunk 4]
```diff
@@ -397,7 +397,7 @@
 				title = kWidget.sanitize( title );
 				description = kWidget.sanitize( description );
 
-				var thumbnailUrl = item.thumbnailUrl || customData.thumbUrl || this.getThumbUrl(item);
+				var thumbnailUrl = this.getThumbUrl(item, customData.thumbUrl);
```
Comment (OrenMe) [inline modules/KalturaSupport/components/playlistAPI.js:400]: if you are passing the custom url then in KBaseMediaList.js you need to change the if condition for the custom asset!

**Pull request merged**